src/google/protobuf/compiler/cpp/cpp_message.cc

```diff
@@ -1217,6 +1217,11 @@ GenerateDescriptorDeclarations(io::Printer* printer) {
       }
     }
 
+    printer->Print("#ifdef _MSC_VER" "\n"
+                   "  $name$OneofInstance() {}" "\n"
```

Comment:
Why would VC require a constructor? I tested v3.0.0-alpha-1 on visual studio 2008 and 2013 but did not find such a problem.